### PR TITLE
[Finishes #99188262] cond blocks for each type

### PR DESCRIPTION
**Pull request by Bjvanminnen**
Each type gets their own conditional block instead of having a generic block one. We will still need to go through levels that use these and replace the existing one with these.

image

### CODE REVIEW

File: dashboard/app/models/studio.rb [hunk 1]
```diff
@@ -341,6 +341,7 @@ def common_blocks(type)
 </category>
 <category name ="Functional Cond">
   <block type="functional_cond" />
+  <block type="functional_cond_number1" />
```
Comment (bcjordan) [inline dashboard/app/models/studio.rb]: s/`functional_cond_number1`/`functional_cond_number`?

File: apps/src/sharedFunctionalBlocks.js [hunk 2]
```diff
@@ -402,13 +421,17 @@ function installCond(blockly, generator) {
...
         .setInline(true);
-      defaultInput.setHSV.apply(defaultInput, Blockly.FunctionalTypeColors.None);
+      defaultInput.setHSV.apply(defaultInput, Blockly.FunctionalTypeColors[type]);
 
       this.appendDummyInput('PLUS')
         .appendTitle(plusField)
         .setInline(true);
 
-      this.setFunctionalOutput(true);
+      if (type !== Blockly.BlockValueType.NONE) {
```
Comment (bcjordan) [inline apps/src/sharedFunctionalBlocks.js]: Should this operator be flipped? i.e., if there _is_ `===` no type, omit the type check

**Comment by bcjordan** [comment]
Just a couple of Qs, LGTM overall though! Colorful :)

File: apps/src/sharedFunctionalBlocks.js (hunk 2 above)
Comment (Bjvanminnen) [inline apps/src/sharedFunctionalBlocks.js]: Haha, yep. Nice catch.

**Pull request merged**